Python/import.c

```diff
@@ -1744,10 +1751,461 @@ import_find_and_load(PyThreadState *tstate, PyObject *abs_name)
     return mod;
 }
 
+static int
+lazy_loaded_contains_parent(PyObject *module, PyObject *name)
+{
+    assert(module != NULL);
+    PyObject *lazy_submodules = PyObject_GetAttr(module, &_Py_ID(__lazy_submodules__));
+    if (lazy_submodules == NULL) {
+        PyErr_Clear();
+        return 0;
+    }
+    int res = PySet_Contains(lazy_submodules, name);
+    Py_DECREF(lazy_submodules);
+    if (res < 0) {
+        return -1;
+    }
+    return res;
+}
+
+static int
+lazy_loaded_add_parent(PyObject *module, PyObject *name)
```

Comment:
and maybe `add_lazy_submodule` here?

Reply:
That's what I was thinking after reading your previous comment. I love it!